datalad/support/annexrepo.py

```diff
@@ -544,22 +536,6 @@ def _check_git_annex_version(cls):
             raise OutdatedExternalDependency(ver_present=ver, **exc_kwargs)
         cls.git_annex_version = ver
 
-    @classmethod
-    def check_direct_mode_support(cls):
```

Comment:
I feel that it was left for a purpose of us being able to at least detect mode support and datalad spitting out DirectModeNoLongerSupportedError in case of running into a such repo. That was the test you are removing about. We no longer want/need that ? How should datalad behave if stepping into a direct mode repo with some older supported git annex?

Reply:
I believe the actual checker `is_direct_mode()` is untouched and functional.

Reply:
> I feel that it was left for a purpose of us being able to at least detect mode support and datalad spitting out DirectModeNoLongerSupportedError in case of running into a such repo.

As @mih mentioned, aborting with DirectModeNoLongerSupportedError still works and is tested.

There are two parts to test_direct_cfg:

  1) check that trying to initialize an annex repo in direct mode reports that DataLad no longer supports it
  2) check that we can call `_set_direct_mode(True)` and get into a direct mode repo despite DataLad not supporting it

That second part fails if git-annex itself doesn't support direct mode.  That's why check_direct_mode_support was restored in af94ddc36 (ENH: annexrepo: Add check_direct_mode_support method, again, 2019-09-06).

>  That was the test you are removing about. We no longer want/need that ? How should datalad behave if stepping into a direct mode repo with some older supported git annex?

As mentioned above, the part of the test being removed check that we can force direct mode with `_set_direct_mode(True)`.  I don't think that is a behavior that we need to test (or even necessarily should), especially give the latest git-annex has removed direct mode entirely.  So I'm fine with deleting part 2 of test_direct_cfg and pruning the then unused check_direct_mode_support.